### PR TITLE
AMQP-92: Replace Basic.Recover with Basic.Reject to avoid unknown del…

### PR DESCRIPTION
**Pull request by facundovs**
…ivery tag issues

### CODE REVIEW

File: mule-transport-amqp/src/main/java/org/mule/transport/amqp/internal/transaction/AmqpTransaction.java [hunk 1]
```diff
@@ -105,43 +112,49 @@ protected void doRollback() throws TransactionException
...
+        Long deliveryTag = getDeliveryTagOrFail(currentMessage, NACK_CHANNEL_ACTION);
         try
         {
             switch (recoverStrategy)
             {
                 case NONE :
-                    // NOOP
+                    try
+                    {
+                        channel.txRollback();
```
Comment (fsgonz) [inline mule-transport-amqp/src/main/java/org/mule/transport/amqp/internal/transaction/AmqpTransaction.java]: Only in the case of NONE a rollback is needed? There are other transactional operations that me be involved in the transaction (for example using the AMQP requester) that may be committed even if the transaction fails.

File: mule-transport-amqp/src/it/resources/exception-strategy-tests-model.json [hunk 2]
```diff
@@ -2,19 +2,27 @@
...
 		"name": "amqpRejectingExceptionStrategy-queue",
 		"durable": false,
-		"auto_delete": true
+		"auto_delete": false
 	},{
 		"name": "amqpTransactedRedeliveryWithRollbackExceptionStrategy-queue",
 		"durable": false,
-		"auto_delete": true
+		"auto_delete": false
+	},{
```
Comment (fsgonz) [inline mule-transport-amqp/src/it/resources/exception-strategy-tests-model.json]: Why are these changes needed in the model used by the test?

Comment (facundovs) [reply mule-transport-amqp/src/it/resources/exception-strategy-tests-model.json]: Since the test case (that contains more than a test) uses different queues that are deleted after the first test is executed. Then when the following tests are executed, the endpoint initialization fails since the queues don't exist anymore .

Comment (fsgonz) [reply mule-transport-amqp/src/it/resources/exception-strategy-tests-model.json]: Ok!
